Source/Core/DolphinQt/HotkeyScheduler.cpp

```diff
@@ -139,10 +139,22 @@ void HotkeyScheduler::Run()
 
   while (!m_stop_requested.IsSet())
   {
-    Common::SleepCurrentThread(1000 / 60);
+    const u64 freelook_update_freq = 200;
+    const u64 interval = SystemTimers::GetTicksPerSecond() / freelook_update_freq;
+    const u64 now = CoreTiming::GetTicks();
```

Comment:
CoreTiming::GetTicks() read might not be atomic/safe on all architectures? Why don't you hook this inside the emulation thread given it depends on emulation timings? It's weird it's in the hotkeys thread if then it needs to depend on the current game speed. I guess it's good for performance, but if this only update inputs, I doubt it would make a relevant difference.

Reply:
FreeLook used to be entirely managed via hotkeys and therefore was on the hotkey thread.  In order to keep things simple in the review where I made FreeLook a proper controller, I kept it on the hotkey thread.  I plan on reworking this in a future PR.